ui/dashboards/src/stories/decorators/WithDatasourceStore.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { StoryFn, StoryContext } from '@storybook/react';
+import { DatasourceStoreProvider, DatasourceStoreProviderProps } from '@perses-dev/dashboards';
+import { GlobalDatasource } from '@perses-dev/core';
+import { EMPTY_DASHBOARD_RESOURCE } from './constants';
+
+export type WithDatasourceStoreParameter = {
+  props: Partial<DatasourceStoreProviderProps>;
+};
+
+// Type guard because storybook types parameters as `any`
+function isWithDatasourceStoreParameter(
+  parameter: unknown | WithDatasourceStoreParameter
+): parameter is WithDatasourceStoreParameter {
+  return !!parameter && typeof parameter === 'object' && 'props' in parameter;
+}
+
+const prometheusDemoUrl = 'https://prometheus.demo.do.prometheus.io';
```

Comment:
Is it ok for us to point Storybook directly at the demo api? I think we're functionally doing something similar (but with the backend doing some proxying) in the Perses app. Tagging @Nexucis as the person most likely to know the answer to this.